p2p/src/authenticated/discovery/actors/listener.rs

```diff
@@ -158,12 +158,11 @@ impl<E: Spawner + Clock + ReasonablyRealtime + Network + Rng + CryptoRng + Metri
             .expect("failed to bind listener");
 
         // Loop over incoming connections as fast as our rate limiter allows
-        let mut shutdown = self.context.stopped();
         let mut accepted = 0;
         select_loop! {
-            _ = &mut shutdown => {
+            self.context,
+            on_shutdown => {
                 debug!("context shutdown, stopping listener");
-                break;
```

Comment:
I suppose it is "obvious" you don't need to break here anymore (although doing so won't cause any issues)?